packages/shared/lib/core/profile-manager/api/events/newOutput.ts

```diff
@@ -1,3 +1,24 @@
-import { OutputData } from '@iota/wallet'
+import { syncBalance } from '@core/account/actions/syncBalance'
+import { activeAccounts } from '@core/profile/stores'
+import { Activity } from '@core/wallet/classes/activity.class'
+import { addActivityToAccountActivitiesInAllAccountActivities } from '@core/wallet/stores/all-account-activities.store'
+import { Bech32Helper } from '@lib/bech32Helper'
+import { Converter } from '@lib/converter'
+import { get } from 'svelte/store'
+import { NewOutputEvent } from '../types/newOutputEvent'
 
-export function handleNewOutputEvent(payload: OutputData): void {}
+export function handleNewOutputEvent(event: NewOutputEvent): void {
+    for (const account of get(activeAccounts)) {
+        const address =
+            event.output.address?.type === 0
+                ? Bech32Helper.toBech32(0, Converter.hexToBytes(event.output.address.pubKeyHash.substring(2)), 'rms')
+                : ''
+        if (event.output.address.type === 0 && account.depositAddress === address && !event.output.remainder) {
+            syncBalance(account.id)
+            addActivityToAccountActivitiesInAllAccountActivities(
+                account.id,
+                new Activity().setFromOutput(event.output.outputId, event.output, account.depositAddress, false, false)
```

Comment:
I wonder if this would more readable if we passed an object in, so we know what the parameters are? Although I have an extension that adds param names anyway.